packages/create-project/src/index.ts

```diff
@@ -1,151 +1,181 @@
 #!/usr/bin/env node
-import os from 'os';
+import chalk from 'chalk';
+import Commander from 'commander';
 import path from 'path';
-import { mkdir, writeFile, readFile } from 'fs-extra';
-import arg from 'arg';
-import { spawn } from './subprocess';
-
-const command = '@temporalio/create';
-const typescriptVersion = '4.4.2';
-const nodeMajorVersion = parseInt(process.versions.node, 10);
-const npm = /^win/.test(process.platform) ? 'npm.cmd' : 'npm';
-
-const packageJsonBase = {
-  version: '0.1.0',
-  private: true,
-  scripts: {
-    build: 'tsc --build',
-    'build.watch': 'tsc --build --watch',
-    start: 'ts-node src/worker.ts',
-    'start.watch': 'nodemon src/worker.ts',
-    workflow: 'ts-node src/exec-workflow.ts',
-  },
-  devDependencies: {
-    typescript: `^${typescriptVersion}`,
-    [`@tsconfig/node${nodeMajorVersion}`]: '^1.0.0',
-    'ts-node': '^10.2.1',
-    nodemon: '^2.0.12',
-  },
-  nodemonConfig: {
-    watch: ['src'],
-    ext: 'ts',
-    execMap: {
-      ts: 'ts-node',
-    },
-  },
-};
-
-const tsConfig = {
-  extends: `@tsconfig/node${nodeMajorVersion}/tsconfig.json`,
-  version: typescriptVersion,
-  compilerOptions: {
-    emitDecoratorMetadata: false,
-    experimentalDecorators: false,
-    declaration: true,
-    declarationMap: true,
-    sourceMap: true,
-    composite: true,
-    rootDir: './src',
-    outDir: './lib',
-  },
-  include: ['src/**/*.ts'],
-  exclude: ['node_modules'],
-};
-
-/**
- * Copy sample from `source` to `target` stripping away snipsync comments
- */
-async function copySample(source: string, target: string) {
-  const code = await readFile(source, 'utf8');
-  const stripped = code.replace(/.*@@@SNIP(START|END).*\n/gm, '');
-  await writeFile(target, stripped);
-}
+import prompts from 'prompts';
+import checkForUpdate from 'update-check';
+import { createApp } from './create-project';
+import { shouldUseYarn } from './helpers/should-use-yarn';
+import { validateNpmName } from './helpers/validate-pkg';
+import packageJson from '../package.json';
 
-async function writePrettyJson(path: string, obj: any) {
-  await writeFile(path, JSON.stringify(obj, null, 2) + os.EOL);
-}
+let projectPath = '';
 
-class UsageError extends Error {
-  public readonly name: string = 'UsageError';
-}
+const program = new Commander.Command(packageJson.name)
+  .version(packageJson.version, '-v, --version')
+  .arguments('<project-directory>')
+  .usage(`${chalk.green('<project-directory>')} [options]`)
+  .action((name) => {
+    projectPath = name;
```

Comment:
Is there a way to do this without this weird global variable?
I've used [`arg`](https://www.npmjs.com/package/arg) before, it's less magic and is dependency free.